File: tests/fixtures/go_project/service.go
package main

// CreateUser creates a new User with the given name.
func CreateUser(name string) User {
	return User{Name: name}
}

// FindUser finds a user by ID (stub for fixture).
func FindUser(id string) *User {
	if id == "1" {
		u := CreateUser("Alice")
		u.Email = "alice@example.com"
		return &u
	}
	return nil
}

// GreetUser returns a greeting for a user by ID.
func GreetUser(id string) string {
	user := FindUser(id)
	if user == nil {
		return "User " + id + " not found"
	}
	return user.Greet()
}

func main() {}

File: tests/fixtures/typescript_project/src/service.ts
import { User } from "./models";
import type { UserOptions, UserId } from "./models";

/**
 * Creates a new User instance.
 */
export function createUser(options: UserOptions): User {
  return new User(options);
}

/**
 * Finds a user by their ID (stub implementation for fixture).
 */
export function findUser(id: UserId): User | null {
  if (id === "1") {
    return createUser({ name: "Alice", email: "alice@example.com" });
  }
  return null;
}

/**
 * Returns a greeting for a user by ID.
 */
export function greetUser(id: UserId): string {
  const user = findUser(id);
  if (!user) return `User ${id} not found`;
  return user.greet();
}
